### ISSUE TITLE
roboter fails when installed with react-styleguidist

### ISSUE THREAD

**Issue by stheine** [thenativeweb/roboter#35]
here's my package json:
```
{
  "name": "test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "fqdn-multi": "0.1.1",
    "react": "15.4.2",
    "react-dom": "15.4.2",
    "react-styleguidist": "4.6.3",
    "roboter": "0.14.4",
    "roboter-client": "0.14.4",
    "webpack": "2.2.1"
  }
}
```

and the roboter.js:
```
'use strict';

const path    = require('path');

const roboter = require('roboter');
const fqdn    = require('fqdn-multi');

roboter
  .workOn('client')
  .equipWith(task => {
    task('universal/analyze', {
      src: ['src/**/*.js', 'src/**/*.jsx'],
      rules: '.eslintrc.json'
    });

    task('universal/test-units', {
      babelize: true,
      src: ['test/units/*Tests.js']
    });

    task('client/watch-app', {
      babelize: [
        path.join(process.cwd(), 'src/')
      ],
      host: fqdn(),
      port: 8888
    });

    task('client/build-app', {
      babelize: [
        path.join(process.cwd(), 'src/')
      ]
    });
  })
  .start();
```

when calling `bot`, this fails with the error:
```
/home/stheine/js/test/node_modules/roboter/lib/tasks/client/build-app/index.js:67
      new webpack.optimize.OccurenceOrderPlugin(),
      ^

TypeError: webpack.optimize.OccurenceOrderPlugin is not a constructor
    at getWebpackConfiguration (/home/stheine/js/test/node_modules/roboter/lib/tasks/client/build-app/index.js:67:7)
    at buildApp (/home/stheine/js/test/node_modules/roboter/lib/tasks/client/build-app/index.js:85:32)
    at Object.keys.filter.forEach.taskName (/home/stheine/js/test/node_modules/roboter/lib/roboter.js:75:7)
    at Array.forEach (native)
    at Roboter.start (/home/stheine/js/test/node_modules/roboter/lib/roboter.js:68:5)
    at Object.<anonymous> (/home/stheine/js/test/roboter.js:35:4)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
```

**Comment by stefanheine** [comment]
as a workaround, I downgrade webpack to 1.13.3 - the same version that roboter-client has in it's package.json. that way, both, roboter and react-styleguideist work.

**Comment by goloroden** [comment]
Can you please provide the exact steps (or even better, an uploaded example) to reproduce this error?

**Comment by stheine** [comment]
```
mkdir <new directory>
cd <new directory>
create the two files listed above (package.json & roboter.js)
npm install
bot
```

-> error

**Comment by stefanheine** [comment]
I guess this is related to the fact that roboter does not bring a dependency to webpack in it's own package.json - that dependency is only satisfied by the roboter-client, but - since it's not in package.json - the exact version is not checked and roboter simly uses the latest version. due to the react-styleguidist installation brings a newer (2.x) version of webpack, suddenly roboter uses that version.

so I think the request is basically about roboter supporting webpack 2.x.

**Comment by goloroden** [comment]
You are right 😊 

Is there a chance to convince you to create a PR for this 😉?
(Maybe with some chocolate, or something like this…?)

**Comment by stefanheine** [comment]
well, honestly, I know too little of the build and test process of roboter and the related roboter-server and roboter-client components. so how do I test those webpack related tasks? I don't think, that 'npm run test' includes those. how do I run tests for the roboter-client related tasks?

**Comment by goloroden** [comment]
Unfortunately, this needs to be done manually, as we do not have automated tests here 😢

But maybe hold on a second, we are currently preparing another feature for roboter, and we have already discussed updating to webpack 2 soon, so stay tuned…

**Comment by goloroden** [comment]
Fixed with roboter 0.14.11, which now makes use of webpack 2 😊
